server/src/test/java/keywhiz/service/resources/admin/SecretsResourceTest.java

```diff
@@ -99,13 +105,54 @@
               NOW, "creator", NOW, "updater",
               emptyMap, 0));
 
-  AuditLog auditLog = new SimpleLogger();
-
   SecretsResource resource;
 
   @Before
   public void setUp() {
     resource = new SecretsResource(secretController, aclDAO, secretDAO, auditLog);
+
+    when(secretController.builder(any(), any(), any(), anyLong()))
+        .thenReturn(secretBuilder);
+    when(secretBuilder.withDescription(any())).thenReturn(secretBuilder);
+    when(secretBuilder.withMetadata(any())).thenReturn(secretBuilder);
+    when(secretBuilder.withOwnerName(any())).thenReturn(secretBuilder);
+    when(secretBuilder.withType(any())).thenReturn(secretBuilder);
+    when(secretBuilder.create()).thenReturn(secret);
+    when(secretBuilder.createOrUpdate()).thenReturn(secret);
+
+    when(secretController.getSecretById(secret.getId())).thenReturn(Optional.of(secret));
+  }
```

Comment:
The various secret creation tests had copies of the same setup boilerplate. Rather than making another copy I opted to centralize everything up-front in the initialization code.